index.js

```diff
@@ -62,7 +81,73 @@ function createTaskCreatedAt(taskCreatedAt) {
 function createTaskCheckBox() {
   return `<input
             type="checkbox"
-            class="table-box-check-box"
             name="checkbox-done"
           /> `;
 }
+
+// function sortTasks(sortType) {
+//   if (sortType == "byTitle") {
+//     renderTasks(sortTasksByTitle());
+//   } else {
+//     renderTasks(unsortTasks());
+//   }
+// }
+
+// function sortTasksByTitle() {
+//   const collator = new Intl.Collator("en", {
+//     numeric: true,
+//   });
+
+//   return tasks.sort((randomTask1, randomTask2) =>
+//     collator.compare(
+//       randomTask1.title.toUpperCase(),
+//       randomTask2.title.toUpperCase()
+//     )
+//   );
+// }
+
+// function unsortTasks() {
+//   return tasks.sort((task1, task2) => (task1.id > task2.id ? 1 : -1));
+// }
+
+function filterTasksByTitle(text) {
+  renderTasks(tasks.filter((task) => task.title.includes(text)));
+}
+
+function filterTasksByTime(time) {
+  switch (time) {
+    case "today":
+      renderTasks(filterTodayTasks());
+      break;
+    case "last-7-days":
+      renderTasks(filterLastDaysTasks(7));
+      break;
+    case "last-30-days":
+      renderTasks(filterLastDaysTasks(30));
+      break;
+    default:
+      renderTasks(tasks);
+  }
+}
+
+function filterTodayTasks() {
+  let filteredTasks = [];
+
+  tasks.map((task) => {
+    if (new Date(task.createdAt) > new Date().setHours(0, 0, 0, 0)) {
+      filteredTasks.push(task);
+    }
+  });
+  return filteredTasks;
+}
```

Comment:
I haven't tested these codes. Might need some minor changes. (consider this approach for the `filterLastDaysTasks` as well)

```suggestion
function filterTodayTasks() {
  return tasks.map((task) => {
      return new Date(task.createdAt) > new Date().setHours(0, 0, 0, 0) ? task : null;
  }).filter((task) => task);
}
```
or even better:
```suggestion
function filterTodayTasks() {
  return tasks.reduce((acc, task) => {
      if (new Date(task.createdAt) > new Date().setHours(0, 0, 0, 0)) {
        acc = [...acc, task];
      }
  }, []);
}
```